### PR TITLE
Updated price calculation

### PR DESCRIPTION
**Pull request by marco-porru**
**Description**

Changes proposed in this pull request:

- support for worker nodes
- support for cloud manager (premium storage, additional costs like redis)
- conversion rate from Capacity Units to Euro with the possibility to apply a flexible discount
- removed the misleading time consumption
- removed the additional nodes (removed from Kyma 1.0)

**Related issue(s)**

### CODE REVIEW

**Comment by marco-porru** [comment]
Screenshot 2024-11-19 at 12 33 15
Screenshot 2024-11-19 at 12 33 33
Screenshot 2024-11-19 at 12 33 45
Screenshot 2024-11-19 at 12 33 52

File: src/calculatorFunctions/totalCosts/calculateTotalCosts.ts [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
+interface Costs {
+  CU: number;
+  CC: number;
 }
+
+export default function calculateTotalCosts(props: Props) : Costs {
+  const { baseConfigCosts, storageCosts, additionalCosts, conversionRatio} = props;
+  const costInCU = baseConfigCosts + storageCosts + additionalCosts;
+  return { CU: costInCU, CC: costInCU * conversionRatio};
+}
```
Comment (mrCherry97) [inline src/calculatorFunctions/totalCosts/calculateTotalCosts.ts:17]: EOF

File: src/state/costs/additionalCostsState.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { atom, RecoilState } from 'recoil';
+import config from '../../config.json';
+
+export const additionalCostsState: RecoilState<number> = atom<number>({
+  key: 'additionalCostsState',
+  default: config.AdditionalConfig.Default,
+});
```
Comment (mrCherry97) [inline src/state/costs/additionalCostsState.ts:7]: EOF

**Review by mrCherry97 — CHANGES_REQUESTED** [review]
One of the Unit tests is failing after changes, please take a look and adjust.

File: src/components/PageLayout/DetailsTable.tsx [hunk 3]
```diff
@@ -26,14 +27,17 @@ export default function DetailsTable() {
...
-          A newly setup Kyma runtime without deployments typically reserves
-          around 8vCPUs for the different components of Kyma.
+          Each single managed Kyma module occupy in-cluster resources. See <a href="https://help.sap.com/docs/btp/sap-business-technology-platform/kyma-modules-sizing" target="_blank" rel="noreferrer">Module Sizing</a> for more info.
         </TableCell>
       </TableRow>
       <TableRow>
         <TableCell>
-          Depending of the setup of your deployments, this can increase.
-          Remaining resources can be consumed for custom workload.
+          Costs are provided in Capacity Units. Use the conversion factor, which incapsulate your current discount (if any), in the Wizard to obtain the correct estimate in Euro.
```
Comment (chriskari) [inline src/components/PageLayout/DetailsTable.tsx:35]: `incapsulates` or I believe more commonly used `encapsulates`

File: src/components/ResultStatistics/CostList.css [hunk 4]
```diff
@@ -5,8 +5,13 @@
...
 }
 
-#final-row {
+.final-row {
   border-bottom: none;
+  display: flex;
+  align-items: center;
+}
+.final-row-child {
+  padding-left: 10px;
```
Comment (chriskari) [inline src/components/ResultStatistics/CostList.css:14]: `final-row-child` is unused and can be removed

File: src/components/UserInputs/storage/GBPremiumStorageInputField.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+import { GBQuantityState } from '../../../state/storage/GBQuantityState';
+import { premiumGBQuantityState } from '../../../state/storage/premiumGBQuantityState';
+import calculateStorageCosts from '../../../calculatorFunctions/storageCosts/calculateStorageCosts';
+import { additionalCostsState } from '../../../state/costs/additionalCostsState';
+import { baseConfigCostsState } from '../../../state/costs/baseConfigCostsState';
+import { applyConversionRateState}  from '../../../state/additionalConfig/applyConversionRateState';
+import { totalCostsInCCState, totalCostsState } from '../../../state/costs/totalCostsState';
+import calculateTotalCosts from '../../../calculatorFunctions/totalCosts/calculateTotalCosts';
+
+export default function TimeConStorageInput() {
```
Comment (chriskari) [inline src/components/UserInputs/storage/GBPremiumStorageInputField.tsx:16]: update the name of this component for clarity

File: src/components/ResultStatistics/CostList.css (hunk 4 above)
Comment (marco-porru) [reply src/components/ResultStatistics/CostList.css:14]: forgotten to commit the file which was using it

**Pull request merged**